coralproject/talk
# [CORL-2859]: Custom flair badges

## What does this PR do?

This adds support for custom flair badges to be passed through via JWT claim for SSO tokens. These custom flair badges are enabled and configured in the admin. When enabled and configured, they are displayed beside a user's username in the stream.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

none

## Does this PR introduce any new environment variables or feature flags?

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can test this PR by going into the admin and enabling custom flair badges (should be disabled by default) under General configuration. Add image urls for custom flair badges. The `Add` button should only be enabled when a valid image url is added. Once the image url is added, it should be shown in the table with a preview for the image. If there is no image (i.e. the link is incorrect or not working), the default broken image icon should display. See that if you try to add an image url that's already added, you receive a helpful duplicate custom flair badge error and can't add it. See that you can delete flair badges.

You can use multi site test to test out flair badges getting added next to usernames in the stream. I updated it so that configured SSO users can also have badges passed through. Add your image url there for your desired SSO users. See the badge then shown in the stream next to that user's username when they add a comment.

Inspect the badge and see that it has the stable class name that is also reflected now in `classes.ts` for flair badges.

## Where any tests migrated to React Testing Library?



## How do we deploy this PR?
